src/engine/enginebuffer.cpp

```diff
@@ -1136,7 +1151,7 @@ void EngineBuffer::processSlip(int iBufferSize) {
     bool enabled = static_cast<bool>(load_atomic(m_slipEnabled));
     if (enabled != m_bSlipEnabledProcessing) {
         m_bSlipEnabledProcessing = enabled;
-        if (enabled) {
+        if (enabled || m_slipCancelled.fetchAndStoreAcquire(0)) {
```

Comment:
I think that's better than the amount logic necessaary to avoid other race conditions.  I'd prefer to miss an extremely short press than get a stuck press.